lib/Wikia/tests/Util/SamplerProxyTest.php

```diff
@@ -0,0 +1,297 @@
+					abs( $count * ( 100 - $samplingRate ) / 100 -
+					     $originalCallableRecorder->getInvocationCount() ) );
+			}
+		}
+	}
+
+	function testResultsCallable() {
+		$testShadowing = true;
+		$testMethodSamplingRate = 100;
+
+		$testOriginalCallable = [ $this->originalMock, $this->originalMethodToSample ];
+		$testAlternateCallable = [ $this->alternateMock, $this->alternateMethod ];
+		$resultsCallable = [ $this->alternateMock, $this->compareResultsMethod ];
+
+		$testArg1 = 1;
+		$testArg2 = 'two';
+		$testArg3 = array( 4, 5, 6 );
+		$originalTestResult = 'original results';
+		$alternateTestResult = 'alternate results';
+
+		$builder = SamplerProxy::createBuilder();
+		$samplerProxy =
+			$builder->setEnableShadowing( $testShadowing )
+				->setMethodSamplingRate( $testMethodSamplingRate )
+				->setOriginalCallable( $testOriginalCallable )
+				->setAlternateCallable( $testAlternateCallable )
+				->setResultsCallable( $resultsCallable )
+				->build();
+
+		$this->originalMock->expects( $this->once() )
+			->method( $this->originalMethodToSample )
+			->with( $testArg1, $testArg2, $testArg3 )
+			->willReturn( $originalTestResult );
+
+		$this->alternateMock->expects( $this->once() )
+			->method( $this->alternateMethod )
+			->with( $testArg1, $testArg2, $testArg3 )
+			->willReturn( $alternateTestResult );
+
+		$this->alternateMock->expects( $this->once() )
+			->method( $this->compareResultsMethod )
+			->with( $originalTestResult, $alternateTestResult )
+			->willReturn( $alternateTestResult );
+
+		$result = $samplerProxy->methodToSample( $testArg1, $testArg2, $testArg3 );
+
+		$this->assertEquals( $alternateTestResult, $result );
+	}
+
+	function testExceptionFallbackFromShadow() {
+		$testShadow = false;
+		$testMethodSamplingRate = 100;
+
+		$testOriginalCallable = [ $this->originalMock, $this->originalMethodToSample ];
+		$testAlternateCallable = [ $this->alternateMock, $this->alternateMethod ];
+
+		$testArg1 = 1;
+		$testArg2 = 'two';
+		$testArg3 = array( 4, 5, 6 );
+		$originalTestResult = 'original test result';
+
+		$builder = SamplerProxy::createBuilder();
+		$samplerProxy =
+			$builder->setEnableShadowing( $testShadow )
+				->setMethodSamplingRate( $testMethodSamplingRate )
+				->setOriginalCallable( $testOriginalCallable )
+				->setAlternateCallable( $testAlternateCallable )
+				->build();
+
+		$this->alternateMock->expects( $this->once() )
+			->method( $this->alternateMethod )
+			->with( $testArg1, $testArg2, $testArg3 )
+			->willThrowException( new \Exception( 'test exception' ) );
+
+		$this->originalMock->expects( $this->once() )
+			->method( $this->originalMethodToSample )
+			->with( $testArg1, $testArg2, $testArg3 )
+			->willReturn( $originalTestResult );
+
+
+		$result = $samplerProxy->methodToSample( $testArg1, $testArg2, $testArg3 );
+
+		$this->assertEquals( $originalTestResult, $result );
+	}
+
+}
+
+class OriginalPopo {
+
+	function methodToSample( $arg1, $arg2, $arg3 ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameters "$arg1, $arg2, $arg3". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)